receiver.go

```diff
@@ -53,30 +54,46 @@ func newReceiver(context *Context, conn *websocket.Conn) *receiver {
 	}
 }
 
-func (r *receiver) receiveLoop() error {
+func (r *receiver) getParseError() error {
+	r.parseErrorLock.Lock()
+	defer r.parseErrorLock.Unlock()
+	return r.parseError
+}
+
+func (r *receiver) setParseError(err error) {
+	r.parseErrorLock.Lock()
+	defer r.parseErrorLock.Unlock()
+	r.parseError = err
+}
+
+func (r *receiver) receiveLoop() (err error) {
 	defer func() {
 		atomic.AddInt32(&r.activeGoroutines, -1)
 	}()
 	go r.parseLoop()
 
 	atomic.AddInt32(&r.activeGoroutines, 1)
 	for {
+		if err = r.getParseError(); err != nil {
+			break
+		}
+
```

Comment:
Can a parse error occur between this check and when we return from websocket.Message.Receive, and not get returned?

Reply:
Yep, that's a fair point if nothing else comes over the websocket. Will take another look.